src/libcrun/linux.c

```diff
@@ -1312,11 +1249,13 @@ do_mounts (libcrun_container_t *container, int rootfsfd, const char *rootfs, lib
           if (strcmp (type, "tmpfs") != 0)
             return crun_make_error (err, 0, "tmpcopyup can be used only with tmpfs");
 
-          copy_from_fd = openat (rootfsfd, target, O_DIRECTORY);
+          copy_from_fd = safe_openat (rootfsfd, rootfs, rootfs_len, target, O_DIRECTORY | O_CLOEXEC, 0, err);
           if (UNLIKELY (copy_from_fd < 0))
             {
               if (errno != ENOTDIR)
-                return crun_make_error (err, errno, "open target for tmpcopyup");
```

Comment:
In general, I think it would be nice to have an ability to amend an error in order to have more context

Reply:
yes I agree, it will be good to have something like `crun_wrap_errorf (`